### DISCUSSION TITLE
Error: no image found for selector "microos-snapshot"

### DISCUSSION THREAD

**Discussion by shadiayoub** [mysticaltech/terraform-hcloud-kube-hetzner#712] [Q&A] [answered]
Hello 
Recently this error keeps popping whenever deploying a new cluster: 
`Error: no image found for selector "microos-snapshot=yes"
│ 
│   with module.kube-hetzner.data.hcloud_image.microos_snapshot,
│   on .terraform/modules/kube-hetzner/main.tf line 6, in data "hcloud_image" "microos_snapshot":
│    6: data "hcloud_image" "microos_snapshot" {`

Any Idea how to work around this?
 Thanks in advance.

**Comment by mysticaltech** [comment]
@shadiayoub This is normal. Read the docs please. All is explained in the readme, you have one command to run to get you going.

**Comment by LuigiClemente** [comment]
I don't mean to be ungrateful, you guys are doing a superb job. I am also here due to the fact that it did create the snapshots but I get the same error. If only the command would be explained it would be helpful to others too I am sure. Alternatively when I mange to figure it out I will happily add in here myself. Thanks

**Comment by LuigiClemente** [comment, marked as answer]
update: 
You need to use the `hcloud context create` command followed by a context name of your choice. This command will prompt you to enter your API token:

```bash
hcloud context create <context-name>
```

Replace `<context-name>` with a name for the context, like "my-hetzner-context" or any other name you prefer. When prompted, enter your API token (`TOKEN`). After setting the context, you can use the `hcloud image list` command to list the images available in your account:

```bash
hcloud image list
```
Now that you have the OpenSUSE MicroOS images available, you can update your Terraform configuration to use these images. In the `main.tf` file of the module (`.terraform/modules/kube-hetzner/main.tf`), locate the `hcloud_image` data source blocks for `microos_x86_snapshot` and `microos_arm_snapshot` and update the `name` attribute to match the names of the images you just listed:

```hcl
data "hcloud_image" "microos_x86_snapshot" {
  name = "OpenSUSE MicroOS x86 by Kube-Hetzner"
}

data "hcloud_image" "microos_arm_snapshot" {
  name = "OpenSUSE MicroOS ARM by Kube-Hetzner"
}
```

After updating the Terraform configuration, you can run the Terraform commands again to create the Kubernetes cluster:

```bash
cd <your-project-folder>
terraform init --upgrade
terraform validate
terraform apply -auto-approve
```

This should resolve the issue, and Terraform should be able to find the required images and create the Kubernetes cluster as expected.

* * *

**Reply by mysticaltech** [reply]
Exactly! Thanks for finding out the path to upgrade to the latest version, appreciate it!
